### ISSUE TITLE
Can it support Intel chip x86 MacBook Pro? Thanks.

### ISSUE THREAD

**Issue by FreeBlues** [divamgupta/diffusionbee-stable-diffusion-ui#153]
Can it support Intel chip x86 MacBook Pro? Thanks.

**Comment by divamgupta** [comment]
Yes!  Check out the releases and let me know if it works or not.

**Comment by ahhyeah** [comment]
I installed the Intel version on a few different Macs and had a couple issues where I get an error that says, "Backend quit unexpectedly." It's working on an iMac Pro, not on an older Mac Pro and then it works on a couple MacBook Pros and not on another MacBook Pro. I thought they were all the same spec laptops so it's weird it doesn't work on one. What info can I share that would be helpful? Thanks! We're having fun as a family with this app! 
![IMG_8526](https://user-images.githubusercontent.com/72570054/193420575-19fb7f45-fa06-4d2b-b2b0-3a3d7e0eb3b9.jpeg)

**Comment by ahhyeah** [comment]
Update, after trying a few times of re-opening the app, it seems to be working... I also did some software updates on a couple and that helped those. The older Mac Pro is running High Sierra and it's working! 😂

**Comment by ahhyeah** [comment]
Curious, will there be a version 1.0 released for intel Macs?

**Comment by FreeBlues** [comment]
@divamgupta @ahhyeah Thanks for your response!

3 days before, I downloaded the 0.6.0 release, and found that it can not run on my MacBook Pro(2020 Intel CPU).
The system said :"You can not open the Diffusion.app, because this Mac does not support the app". 
These days I tried several stable-diffusion repos to run on my MacBook Pro, and 2 of them succeed(only can be run on command line).
Right now I downloaded the 1.0.0 release here, it can not work yet and got the same prompt. I think maybe in this App, only M1/M2'codes were built without Intel's.  


Here is the repos I used, maybe be helpful for this project.

-  [1st one](https://github.com/bfirsh/stable-diffusion.git):

Install steps:

Install `python3`，version >=3.10

```
git clone -b apple-silicon-mps-support https://github.com/bfirsh/stable-diffusion.git macSD
cd macSD
mkdir -p models/ldm/stable-diffusion-v1/
ln -s ~/Github/sd-data/sd-v1-4.ckpt ./models/ldm/stable-diffusion-v1/model.ckpt
```
The `sd-v1-4.ckpt` need to be downloaded from `Hugging Face repository`, and saved into `~/Github/sd-data/sd-v1-4.ckpt`

Setup `python` virtual env `macenv`

```
python3 -m pip install virtualenv
python3 -m virtualenv macenv
```

Activate virtualenv：macenv

```
source macenv/bin/activate
```

After activated, install the depends：

```
pip install -r requirements.txt	
```

All ok! use it in command line:

-	txt2img

```
python scripts/txt2img.py \     
  --prompt "a red juicy apple floating in outer space, like a planet" \
  --n_samples 1 --n_iter 1 --plms --precision full
```

-	img2img

```
python scripts/img2img.py --prompt "A fantasy landscape, trending on artstation" --init-img <path-to-img.jpg> --strength 0.8
```

-  [2nd one](https://github.com/invoke-ai/InvokeAI.git):

Follow this article:

`
https://github.com/invoke-ai/InvokeAI/blob/main/docs/installation/INSTALL_MAC.md
`






@ahhyeah I think your MacBook Pros must are all M1 chip.

**Comment by ahhyeah** [comment]
Diffusionbee 0.3.0 intel version is here on this github under "releases": https://github.com/divamgupta/diffusionbee-stable-diffusion-ui/releases/download/0.3.0/DiffusionBee-0.3.0-intel-macos.dmg

**Comment by FreeBlues** [comment]
> Diffusionbee 0.3.0 intel version is here on this github under "releases": https://github.com/divamgupta/diffusionbee-stable-diffusion-ui/releases/download/0.3.0/DiffusionBee-0.3.0-intel-macos.dmg

Thanks a lot! It can work! very easy to use! Wish to update to the latest.

**Comment by fortinmike** [comment]
@divamgupta Any reason why versions >0.3.0 aren't compatible with Intel Macs? I tried running the app from source and nothing works, even text-to-image which worked fine before. It just generates a test image.

**Comment by FreeBlues** [comment]
> @divamgupta Any reason why versions >0.3.0 aren't compatible with Intel Macs? I tried running the app from source and nothing works, even text-to-image which worked fine before. It just generates a test image. I was looking forward to playing with image-to-image (and outpainting!) on my Intel MBP 16.

We also look forward to continuing to support Intel CPU MacBookPro with the new version.

**Comment by divamgupta** [comment]
Yes!  Check out the releases and let me know if it works or not.

**Comment by peternoordijk** [comment]
@divamgupta I did actually check the v1.0.0 arm64.dmg file last week on my intel macbook pro, just to be sure, but it obviously doesn't work. Meanwhile, there's also a new issue reported specifically for this https://github.com/divamgupta/diffusionbee-stable-diffusion-ui/issues/182

**Comment by fortinmike** [comment]
@divamgupta Is this a mistake then? Were Intel builds disabled by mistake? Because all releases > 0.3.0 do not contain an Intel dmg. Only arm64.

**Comment by fortinmike** [comment]
Seems like the Intel build came back for 1.4.2 but was again removed in 1.4.3. @divamgupta mentioned issues with AMD GPUs which I assume might be part of the reason why the Intel build was removed again. Seems like the 1.4.2 build was set to use GPU even on AMD, which caused black images to be generated on my machine.

However, out of curiosity (and mostly because 0.3.0 worked fine on my Intel Mac with AMD GPU), I tried to build the app from source. To my surprise, it works just fine. It does run on the CPU, as before, but that's fine with me. Better than nothing.

To do the same on your machine, do the following:

1. Install node.js 16.17.1 (LTS)
2. Open Terminal, switch to the directory where you want the source to be downloaded `cd ~/Downloads` (for example) and run `git clone https://github.com/divamgupta/diffusionbee-stable-diffusion-ui.git`
3. This one is a quick hack, but without time to research the source further, it works as a proof of concept. In the cloned source, open the `electron_app/src/bridge.js` file and replace line 16 (`let script_path = ...`) with this: `let script_path = "../backends/stable_diffusion_tf/diffusionbee_backend.py"; ` (@divamgupta can you comment on the correct way to get rid of the fake backend and use the real one instead?)
4. In Terminal, cd to the `electron_app` directory and run `npm run electron:serve`.

This will start the app, which AFAIK works just fine. I tried text-to-image, inpainting, image-to-image, etc. I did not encounter issues so far. So, why does the Intel build keep being removed? Are there some subtle issues which are deal-breakers on Intel hardware? I didn't notice any so far. It'd be great to be able to use official release builds instead of this workaround.

![Screenshot 2022-10-29 at 11 18 56](https://user-images.githubusercontent.com/118257/198839526-3d1804f7-23c4-44fd-82ac-d7ca33e8028f.png)

**Comment by divamgupta** [comment]
Hey, thanks @fortinmike  , would you be up for helping in getting the metal optimized version of Bee for AMD GPUs. could you email me at divamg@alumni.cmu.edu

**Comment by sascha1337** [comment]
just need to change the deployment target to universal in xcode, maybe

**Comment by fortinmike** [comment]
@divamgupta Nice to hear you're working on a Metal version that could run on AMD GPUs! Code-wise, ML things are far from my area of expertise, though (if coding help is what you had in mind). I'm not sure I would be of much help on this front, but I can test alpha builds on my hardware if you don't have an Intel / AMD Mac on hand. I sent you an email.
